src/python/tmclient/api.py

```diff
@@ -2024,40 +2050,26 @@ def upload_feature_values(self, mapobject_type_name, plate_name,
             x-position of the site relative to the well grid
         tpoint: int
             zero-based time point index
-        data: pandas.DataFrame
-            *n*x*p* dataframe, where *n* are number of objects at this site
-            and *p* number of features (index must be site-specific one-based
-            labels that must match those of the corresponding segmentation
-            image)
+        filename: str
+            path to the file on disk
+        index_col: str
+            column name containing the object labels
 
         See also
         --------
         :func:`tmserver.api.feature.add_feature_values`
         :class:`tmlib.models.feature.FeatureValues`
         '''
-        logger.info(
-            'upload feature values for experiment "%s", object type "%s" at '
-            'plate "%s", well "%s", y %d, x %d, tpoint %d',
-            self.experiment_name, mapobject_type_name, plate_name, well_name,
-            well_pos_y, well_pos_x, tpoint
-        )
-        content = {
-            'plate_name': plate_name, 'well_name': well_name,
-            'well_pos_y': well_pos_y, 'well_pos_x': well_pos_x,
-            'tpoint': tpoint,
-            'names': data.columns.tolist(),
-            'labels': data.index.tolist(),
-            'values': data.values.tolist()
-        }
-        mapobject_type_id = self._get_mapobject_type_id(mapobject_type_name)
-        url = self._build_api_url(
-            '/experiments/{experiment_id}/mapobject_types/{mapobject_type_id}/feature-values'.format(
-                experiment_id=self._experiment_id,
-                mapobject_type_id=mapobject_type_id
-            )
+        logger.info('upload feature value file "%s"', filename)
+        if not filename.endswith('csv'):
+            raise IOError('Filename must have "csv" extension.')
```

Comment:
This may break on windows, if files have uppercase names.

Reply:
I totally agree - thanks for the hint. I added a fix.